### DISCUSSION TITLE
Adding new props

### DISCUSSION THREAD

**Discussion by lsangreg** [carla-simulator/carla#6788] [Q&A]
Hello everyone. 
I have been trying adding new props in carla using UE4 and https://carla.readthedocs.io/en/0.9.12/tuto_A_add_props/#ingestion-in-a-build-from-source this tutorial.
This is the json: 
```json
{
  "maps": [],
  "props": [
    {
      "name": "Roller",
      "size": "medium",
      "source": "./Props/Prop01/SoilRoller.fbx",
      "tag": "Static"
    }]
}
```
My package is called Roller, when I import it using `make import` I can see it in  UE4 and I have a folder structure like this:
├── Roller
│   ├── Config
│   │   └── Roller.Package.json
│   ├── Maps
│   ├── Static
│   │   ├── Static
│   │   │   ├── Roller
│   │   │   │   └── SoilRoller_SoilRoller.uasset
│   │   │   │   └── Material.uasset

When it comes to export the package as standalone for CARLA, I use command `make package ARGS="--packages=Roller"`  and this is what I obtain, in the tar.gz file:

 ├── Roller_0.9.12-dirty
│   ├── CarlaUE4
│   │   ├── Content
│   │   │   ├── Carla
│   │   │   │   ├── Blueprints
│   │   │   │   ├── Static 
│   │   │   ├── Roller
│   │   │   │   ├── Config
│   ├── Engine

So inside the Roller i have just config, without the Static folder with the uassets and the materials.

I also tried adding these files manually, so I imported them in carla but then I could not find them in the blueprints, as static actor to spawn, which from what I understood,  should be possible.

**Comment by LeonardoVezzani** [comment]
Did you manage to solve it? Were you able to swan the actor client side?

**Reply by lsangreg** [reply]
Yes, the problem is in the name.
It can be solved by changing the name of prop in ueeditor. When you import the model into CarlaUe, the name of the model mesh is automatically added after it. So, just delete any extra names. For example, the  model name is abc and the mesh name is abc, which becomes abc_abc when imported. Just write the fbx name.
(salutami il poli :) )

**Reply by lsangreg** [reply]
before that I aslo made these fixes:

The fix for me is to add after this [line](https://github.com/carla-simulator/carla/blob/0.9.12/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Commandlet/PrepareAssetsForCookingCommandlet.cpp#L514), located in
(carla/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Commandlet line 514):
MapPathDataLinux.Append(PropsMapPath + TEXT("/PropsMap"));
This should also be a new commit in the carla repo, so in case check if you are using the last commit!

There are also modification done in Import.py , check their last commit https://github.com/carla-simulator/carla/commit/29474670ad3cdeba8f733c6ff45d91d4216b779e

**Reply by LeonardoVezzani** [reply]
thank you so much. Do you have other source I can keep on hand while working on it?
I'm trying to create a speedometer for immersive testing in CARLA and it is becoming quite difficult to access the sensors... 😅 
(il poli ricambia ;) )

**Reply by lsangreg** [reply]
Not really. I don't know if you are already doing it, but to access sensors it is good to use the ros-bridge (ROS2) and subscribing to the topic / visualizing them in rviz2

**Reply by LeonardoVezzani** [reply]
Good to know, thank you;)
Just one last question: do your .uasset contain any c++ script with interfaces to the pythonapi?
